### PR TITLE
Successfully implemented RegistrationServiceImpl, RegistrationExcepti…

### PR DESCRIPTION
**Pull request by Bohdan-Halitskyi**
…on and RegistrationServiceImplTest classes.

### CODE REVIEW

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    @Test
+    void validUser_Ok() {
+        registrationService.register(user);
+        User expected = storageDao.get(user.getLogin());
+        Assertions.assertEquals(expected, user, "Age field's passed");
+    }
+
+    @Test
+    void register_passwordLengthIsLessThanMinLength_notOk() {
```
Comment (Dmytro-Mk) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:87]:
```suggestion
    void register_userPasswordLengthIsLessThanMinLength_notOk() {
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        user.setPassword("321");
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "User`s password should be 6 or more symbols");
+    }
+
+    @Test
+    void register_passwordLengthIsLongerThanMinLength_ok() {
```
Comment (Dmytro-Mk) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:95]:
```suggestion
    void register_userPasswordLengthIsLongerThanMinLength_ok() {
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        user.setPassword(DEFAULT_PASSWORD + "1");
+        registrationService.register(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(user, actual, "All user`s fields are correct");
+    }
+
+    @Test
+    void register_loginExists_notOk() {
```
Comment (Dmytro-Mk) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:103]:
```suggestion
    void register_userLoginAlreadyExists_notOk() {
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        registrationService.register(user);
+        registrationService.register(user);
+        Assertions.assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "User is already exists");
+    }
+}
```
Comment (Dmytro-Mk) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:110]: messages in assert are shown when test fails, so you need to show info that helps us understand why test fail and what is expected behavior
ex: "should throw OurException when user is null"
rewrite all messages and make them informative

also you need tests for all edge case scenarious: password length 5/6/7  age 17/18/19

**Review by Dmytro-Mk — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 5]
```diff
@@ -5,10 +5,45 @@
...
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_AGE = 18;
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        userValidation(user);
```
Comment (Dmytro-Mk) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:14]:
```suggestion
        validateUser(user);
```

**Review by Dmytro-Mk — CHANGES_REQUESTED** [review]

**Review by Dmytro-Mk — APPROVED** [review]

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    @Test
+    void register_ageMoreThanValid_OK() {
+        user.setAge(VALID_AGE + 1);
+        registrationService.register(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(user, actual, "User should be added if age is greater than " + VALID_AGE);
+    }
+
+    @Test
+    void register_validUserIsAddedToStorage_Ok() {
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:130]:
```suggestion
    void register_ageIsValid_ok() {
        user.setAge(VALID_AGE);
        registrationService.register(user);
        User actual = storageDao.get(user.getLogin());
        assertEquals(user, actual, "User should be added if age is " + VALID_AGE);
    }

    @Test
    void register_ageMoreThanValid_ok() {
        user.setAge(VALID_AGE + 1);
        registrationService.register(user);
        User actual = storageDao.get(user.getLogin());
        assertEquals(user, actual, "User should be added if age is greater than " + VALID_AGE);
    }

    @Test
    void register_validUserIsAddedToStorage_ok() {
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    void register_userLoginIsNull_notOk() {
+        user.setLogin(null);
+        Assertions.assertThrows(EXCEPTION_CLASS,
+                () -> registrationService.register(user),
+                "Should throw RegistrationException if user login is null");
+    }
+
+    @Test
+    void register_userLoginAlreadyExists_notOk() {
+        registrationService.register(user);
```
Comment (KostyaNikiforov) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:60]: It is better to add user here using storageDao

**Review by KostyaNikiforov — APPROVED** [review]
